test/Microsoft.AspNet.Mvc.FunctionalTests/TryValidateModelTest.cs

```diff
@@ -37,12 +37,7 @@ public class TryValidateModelTest
             // Assert
             var body = await response.Content.ReadAsStringAsync();
             var json = JsonConvert.DeserializeObject<Dictionary<string, string>>(body);
-            Assert.Equal(8, json.Count);
-            Assert.Equal("CompanyName cannot be null or empty.", json["product.CompanyName"]);
-            Assert.Equal("The field Price must be between 20 and 100.", json["product.Price"]);
-            Assert.Equal("The Category field is required.", json["product.Category"]);
-            Assert.Equal("The field Contact Us must be a string with a maximum length of 20." +
-                "The field Contact Us must match the regular expression '^[0-9]*$'.", json["product.Contact"]);
+            Assert.Equal(4, json.Count);
```

Comment:
before this change, validation after model binding added 4 errors with `product` prefix and then a call to `TryValidate` added the 4 errors without prefix. After this change, we no longer get the prefix errors and `TryValidate' clears the errors in ModelState and re-writes them